codegen/templates/manifests/crd.yamltmpl

```diff
@@ -1,4 +1,7 @@
 [[- range $crd := .Crds -]]
+[[- if (gt (len $crd.CustomEnableCondition) 0) ]]
+[[ printf "{{- if %s }}" $crd.CustomEnableCondition ]]
+[[- end ]]
```

Comment:
This is going to be weird: for experimental api's when `CustomEnableCondition` is present we are going to have two conditionals, both of which would need to evaluate to true for them to get installed.

Is there a requirement for having a configurable enable condition instead of using enable/disable lists?

Reply:
I think the use case is a bit different that experimental flag does some checking based of the version. But this work intends to group crds relevant to a specific product. IMO reusing that flag would be weird

Reply:
What I'm trying to say is that for experimental CRDs rendered template is going to look like:

```
{{- if some-condition-here }}
{{- if has "externalworkloads.networking.gloo.solo.io/v2alpha1" $.Values.enabledExperimentalApi }}
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  annotations:
    crd.solo.io/specHash: 9865652aecbb7957
    crd.solo.io/version: 2.5.0
  labels:
    app: gloo-mesh-apis
    app.kubernetes.io/name: gloo-mesh-apis
  name: externalworkloads.networking.gloo.solo.io
spec:
```

and for an experimental CRD above to get enabled `some-condition-here` must be true and `Values.enabledExperimentalApi` must contain `externalworkloads.networking.gloo.solo.io/v2alpha1`, which is probably not what we want.

Reply:
I don't think it would matter that there are two conditions. It wouldn't be a huge deal technically `if and () ()` are the same as `if { if {} }`. So if the CRDs so happens to be experimental then sure it may end up like that. But I think a lot of this will be handled by default values in the chart that is responsible for CRDs